docs/site/todo-list-tutorial-sqldb.md

```diff
@@ -0,0 +1,134 @@
+
+### Create a new datasource
+
+We're going to reuse the DataSource `db` to connect to the PostgreSQL database
+instead of the in-memory database.
+
+First, remove the existing `db`, run:
+
+```sh
+rm src/datasources/db.datasource.*
+```
+
+Then, create the datasource with the same name with the `lb4 datasource` command
+and select `PostgreSQL` as the connector.
+
+```sh
+$ lb4 datasource
+? Datasource name: db
+? Select the connector for db:
+  Redis key-value connector (supported by StrongLoop)
+  MongoDB (supported by StrongLoop)
+  MySQL (supported by StrongLoop)
+❯ PostgreSQL (supported by StrongLoop)
+  Oracle (supported by StrongLoop)
+  Microsoft SQL (supported by StrongLoop)
+  REST services (supported by StrongLoop)
+  ...
+```
+
+### Specify the foreign key constraint in the models
+
+Based on the model relations we've specified in the 3 models, the expected
+database tables look like below:
+![database tables](todo-list-tutorial-dbtables.png)
+
+{% include note.html content="
+There is ongoing work on supporting strong relations with referential integrity.  For details, please see [epic #2231](https://github.com/strongloop/loopback-next/issues/2331).
+" %}
+
+#### Specify the foreign key constraints in Todo model
+
+We will use the `foreignKeys` attribute to determine the constraints in the
+database table.
+
+In `src/models/todo.model.ts`, add the `settings` options in the `@model`
+decorator.
+
+```ts
+@model({
+  settings: {
+    foreignKeys: {
+      fk_todo_todoListId: {
+        name: 'fk_todo_todoListId',
+        entity: 'TodoList',
+        entityKey: 'id',
+        foreignKey: 'todolistid',
+      },
+    },
+  },
+})
+export class Todo extends Entity {
+```
+
+Check out the details for
+[auto-migrate and auto-update](https://loopback.io/doc/en/lb3/PostgreSQL-connector.html#auto-migrateauto-update-models-with-foreign-keys)
+for PostgreSQL connector. For other connectors, go to the corresponding
+connector from the
+[Connectors Reference](https://loopback.io/doc/en/lb3/Connectors-reference.html)
+section.
+
+#### Specify the foreign key constraints in TodoListImage model
+
+Similar to the `Todo` model, we will specify the constraints in the `settings`.
+
+```ts
+@model({
+  settings: {
+    foreignKeys: {
+      fk_todoListImage_todoListId: {
+        name: 'fk_todoListImage_todoListId',
+        entity: 'TodoList',
+        entityKey: 'id',
+        foreignKey: 'todolistid',
+      },
+    },
+  },
+})
+export class TodoListImage extends Entity {
+```
+
+### Database migration using `npm run migrate` command
+
+The order of table creation is important. We are going to migrate `TodoList`
+model before the `Todo` and `TodoListImage` models.
+
+In `src/migrate.ts`, modify this line:
+
+```ts
+await app.migrateSchema({existingSchema});
+```
+
+to:
+
+```ts
+await app.migrateSchema({
+  existingSchema: existingSchema,
+  models: ['TodoList', 'Todo', 'TodoListImage'],
```

Comment:
Are you sure this line is necessary? I believe that `app.migrateSchema` is going to migrate ALL models by default. By omitting the list of models, we keep the code forward-compatible. In your proposal, developers adding new models in the future must remember to update the migration script too.

Reply:
From my experiment, this line is necessary.  `TodoList` table must be created first before `Todo` and `TodoListImage`.  When I didn't specify the order, I will get an error about the TodoList table do not exist.  I'm not sure what's the better way to specify the order but this change seems to be working for me.

Reply:
Discussed with @bajtos.  He suggested: 
1. Add comments in the above code snippet to explain why we need to do this. -- DONE
2. Open a GH issue to investigate how we can improve the migration tool so that users don't need to explicitly specify the order.  -- https://github.com/strongloop/loopback-next/issues/2831 is created.